src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultKeyValueAdapter.cs

```diff
@@ -102,5 +92,53 @@ public bool NeedsRefresh()
         {
             return _secretProvider.ShouldRefreshKeyVaultSecrets();
         }
+
+        private string ParseSecretReferenceUri(ConfigurationSetting setting)
+        {
+            try
+            {
+                var reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(setting.Value));
+
+                if (reader.Read() && reader.TokenType != JsonTokenType.StartObject)
+                {
+                    throw new FormatException(string.Format(ErrorMessages.KeyVaultSecretReferenceInvalidFormat, setting.Key));
```

Comment:
Where are we handling FormatException?

Reply:
We handle it on startup [here](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/4ce9ec53988dc27d3045406601ee9961c5e1cda8/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs#L56)

But in the [TryRefreshAsync error handling code path](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/4ce9ec53988dc27d3045406601ee9961c5e1cda8/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs#L443) it looks like it is unhandled.

Reply:
The first one is caught during build (while creating ConfigurationClient), so we can still fail with FormatException during startup. This would break the "optional" behavior [here](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/4ce9ec53988dc27d3045406601ee9961c5e1cda8/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs#L631).

Reply:
Ah interesting. It seems invalid feature flags break the optional promise, even in released versions.

Reply:
You're right. Didn't realize we were always throwing FormatException for invalid feature flags.

Reply:
So it sounds like I can add the `FormatException` handling to `TryRefreshAsync` then

Reply:
I would open an issue and solve it in a separate PR. The exception propagation/handling isn't changing in this PR.

Reply:
ok, will do